jerry-core/ecma/base/ecma-globals.h

```diff
@@ -227,56 +306,61 @@ typedef struct
 } ecma_getter_setter_pointers_t;
 
 /**
- * Description of ecma-property
+ * Property data.
  */
-typedef struct ecma_property_t
+typedef union
 {
-  /** Compressed pointer to next property */
-  mem_cpointer_t next_property_p;
+  ecma_value_t value; /**< value of a property */
+  ecma_getter_setter_pointers_t getter_setter_pair; /**< getter setter pair */
+} ecma_property_value_t;
 
-  /** Property's flags (ecma_property_flags_t) */
-  uint8_t flags;
+/**
+ * Property pair.
+ */
+typedef struct
+{
+  ecma_property_header_t header; /**< header of the property */
+  ecma_property_value_t first_value; /**< first property value */
+  ecma_property_value_t second_value; /**< second property value */
+  mem_cpointer_t first_name_cp; /**< first property name */
+  mem_cpointer_t second_name_cp; /**< second property name */
```

Comment:
I wonder whether packing together 2 properties is the most efficient approach? Why not 4 or 6? 6 would keep alignments right as well.

I have a probable answer to that myself: having `third_`, `fourth_`, `fifth_`, `sixth_` etc. fields (1) would be ugly and (2) would blow up the code. As I've already seen quite some code duplications and special case handlings below to deal with first and second fields.

I also have a potential treatment for that. Why not use arrays both in `ecma_property_header_t` and here (in `ecma_property_pair_t`) with a constant size? It can be 2, but can also be configured to another number (e.g., to 6). It could also have the benefit to rewrite code dups to loops (e.g., in `ecma_delete_property`, `ecma_op_object_get_property_names`). I have the gut feeling that this approach would keep all the benefits of yours and result in cleaner and easier to maintain code. (I'd guess that the compiler would unroll loops of 2, so performance should not be hindered.)

What do you think about this idea?